### DISCUSSION TITLE
[v2] How to determine whether query is polling?

### DISCUSSION THREAD

**Discussion by SevenOutman** [TanStack/query#1530] [Q&A]
I'm using react-query 2. In my case, I'm writing a table component with polling and here is my query:

```typescript
  const { resolvedData } = usePaginatedQuery(
    ['todo', params],
    () => queryTodos(params),
    {
      refetchInterval: 5000
    }
  );
```

I want to show a loading indicator when my query is fetching due to `params` change but not polling on interval, but can't find an API to tell polling from fetching.

**Comment by TkDodo** [comment]
`isLoading` will be true when params change, but not on refetches of any kind. `isFetching` will always be true when a request is outbound.

**Reply by SevenOutman** [reply]
To be more specific, I want to show the loading indicator even if it's refetching with a params that's already been loaded.

What if I call `refetch` manually? How do I tell polling from a manual refetch?

**Reply by TkDodo** [reply]
I don’t know of such a setting.
